core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -838,7 +840,7 @@ private[spark] class TaskSetManager(
 
     if (!isZombie && reason.countTowardsTaskFailures) {
       taskSetBlacklistHelperOpt.foreach(_.updateBlacklistForFailedTask(
-        info.host, info.executorId, index))
+        info.host, info.executorId, index, Some(failureReason)))
       assert (null != failureReason)
```

Comment:
move the `assert (null != failureReason)` first, and to go along with the other change, drop the `Some` wrapper around `failureReason`.